syntax = "proto3";

service ProductsDataService{
    rpc GetProductCandles(ProductCandleRequest) returns(ProductCandleResponse);
}

message ProductCandleRequest{
    string product_id = 1;
    int32 granularity = 2;
    optional int32 requests = 3;
    optional int32 data_points_limit = 4;
}

message ProductCandleResponse{
    message ProductCandle{
        // bucket start time
        int32 time = 1;
        // lowest price during the bucket interval
        double low = 2;
        // highest price during the bucket interval
        double high = 3;
        // opening price (first trade) in the bucket interval
        double open = 4;
        // closing price (last trade) in the bucket interval
        double close = 5;
        // volume of trading activity during the bucket interval
        double volume = 6;
    }
    repeated ProductCandle product_candles = 1;
}
